packages/dashboard/src/components/composed/Layout.tsx

```diff
@@ -13,10 +25,43 @@ const useStyles = createStyles((_theme, _params, _getRef) => ({
 
 function Layout(): JSX.Element {
   const {
-    state: { notice }
+    state: { notice, analyticsConfig },
+    operations: { updateAnalyticsConfig }
   } = useDash()!;
   const { classes } = useStyles();
 
+  useEffect(() => {
+    if (notice.show || analyticsConfig.analyticsSet === null) return;
+
+    const shouldAsk =
+      !analyticsConfig.enableAnalytics && !analyticsConfig.analyticsSet;
+    const shouldAskAgain =
+      !analyticsConfig.enableAnalytics &&
+      Date.now() - analyticsConfig.analyticsMessageDateTime! >
+        ARBITRARY_ANALYTICS_NEXT_ASK_THRESHOLD;
```

Comment:
Note that this kind of math is rife with edge cases around leap years and whatnot. Better if we can do a proper duration check vs. math and assuming we know how many milliseconds a year is.

Happy to accept an issue for this instead of fixing it now.

Reply:
(OK, if this triggers a day early, it doesn't really matter. But still... good practice not to assume you can just subtract dates like that)

Reply:
https://date-fns.org/ may be a viable option

Reply:
> Happy to accept an issue for this instead of fixing it now.

+1

Reply:
Good point! I've changed this to compare `Date` objects instead. The logic is as follows:
```ts
const askAfter = new Date(lastAskedInMs);
askAfter.setDate(askAfter.getDate() + 365);

if (new Date() > askAfter) // Ask again!
```
Note that this _is_ affected by leap years. We can fix it by using the pair `setFullYear` and `getFullYear` instead of `setDate` and `getDate`, but I figured days are easier to work with if we later want to change this threshold to 90 days or something.

Reply:
If we wanted to be super accurate, the way to do this would be to do, like, `add(date, thresholdDuration)`, where `thresholdDuration = { years: 1 }`, then check against that. 

But it's okay. If you say "no" on 2023-12-01 and we ask again on 2024-11-30... kinda trivial.